src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -148,9 +148,431 @@
+    int minusAdded = false;
+    int digitAdded = false;
+    int currencyAdded = false;
+    int spaceAdded = false;
+
+    for (int i = iStart; i <= iEnd; i++)
+    {
+        switch (srcPattern[i])
+        {
+            case CHAR_MINUS:
+            case CHAR_OPENPAREN:
+            case CHAR_CLOSEPAREN:
+                minusAdded = true;
+                break;
+        }
+
+        if (minusAdded)
+           break;
+    }
+
+    // international currency symbol (CHAR_CURRENCY)
+    // The positive pattern comes first, then an optional negative pattern
+    // separated by a semicolon
+    // A destPattern example: "(C n)" where C represents the currency symbol, and
+    // n is the number
+    char* destPattern;
+
+    // if there is no negative subpattern, prefix the minus sign
+    if (isNegative && !minusAdded)
+    {
+        int length = (iEnd - iStart) + 2;
+        destPattern = (char*)calloc((size_t)length, sizeof(char));
+        if (!destPattern)
+        {
+            return NULL;
+        }
+        destPattern[index++] = '-';
+    }
+    else
+    {
+        int length = (iEnd - iStart) + 1;
+        destPattern = (char*)calloc((size_t)length, sizeof(char));
+        if (!destPattern)
+        {
+            return NULL;
+        }
+    }
+
+    for (int i = iStart; i <= iEnd; i++)
+    {
+        char ch = srcPattern[i];
+        switch (ch)
+        {
+            case CHAR_DIGIT:
+            case CHAR_ZERO:
+                if (!digitAdded)
+                {
+                    digitAdded = true;
+                    destPattern[index++] = 'n';
+                }
+                break;
+
+            case CHAR_CURRENCY:
+                if (!currencyAdded)
+                {
+                    currencyAdded = true;
+                    destPattern[index++] = 'C';
+                }
+                break;
+
+            case CHAR_SPACE:
+            case CHAR_NBSPACE:
+                if (!spaceAdded)
+                {
+                    spaceAdded = true;
+                    destPattern[index++] = ' ';
+                }
+                break;
+
+            case CHAR_MINUS:
+            case CHAR_OPENPAREN:
+            case CHAR_CLOSEPAREN:
+                minusAdded = true;
+                destPattern[index++] = (char)ch;
+                break;
+
+            case CHAR_PERCENT:
+                destPattern[index++] = '%';
+                break;
+        }
+    }
+
+    return destPattern;
+}
+
+/*
+Function:
+GetNumericPattern
+
+Determines the pattern from the decimalFormat and returns the matching pattern's
+index from patterns[].
+Returns index -1 if no pattern is found.
+*/
+static int GetNumericPatternNative(const char* pNumberFormat,
+                                   const char* patterns[],
+                                   int patternsCount,
+                                   int isNegative)
+{
+    const int INVALID_FORMAT = -1;
+    const int MEMORY_ALLOCATION_ERROR = 7;
```

Comment:
I updated this part so it will return INVALID_FORMAT in case it can't find and from C# added this check
 ```
if (value < 0)
 {
        Debug.Fail("[CultureData.GetLocaleInfoNative(LocaleNumberData)] failed");
 }
```

Reply:
Compared what we have now with ICU we don't throw exception https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs#L163 .

Reply:
However, they have non-negative default values if the pattern is not found from the APIs I saw in https://github.com/dotnet/runtime/blob/926062aa4262dcfaf456a85752155da03190b0da/src/native/libs/System.Globalization.Native/pal_localeNumberData.c#L419 (GetNumberNegativePattern, GetCurrencyPositivePattern, GetCurrencyNegativePattern, GetPercentNegativePattern, GetPercentPositivePattern), whereas here, a negative is allowed.

Reply:
Implemented default value in C# side . As now functions are just returning the value I will send negative values to assert in C# and than return default value.